core/src/main/scala/com/salesforce/op/stages/impl/feature/IDVectorizer.scala

```diff
@@ -0,0 +1,170 @@
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.stages.impl.feature
+
+import com.salesforce.op.{FeatureHistory, UID}
+import com.salesforce.op.features.TransientFeature
+import com.salesforce.op.features.types._
+import com.salesforce.op.stages.base.sequence.{SequenceEstimator, SequenceModel}
+import com.salesforce.op.utils.json.JsonLike
+import com.salesforce.op.utils.spark.{OpVectorColumnMetadata, OpVectorMetadata}
+import com.twitter.algebird.Monoid._
+import com.twitter.algebird.Operators._
+import com.twitter.algebird.Semigroup
+import org.apache.spark.sql.{Dataset, Encoder}
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import com.salesforce.op.utils.spark.RichDataset._
+
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * Convert a sequence of id features into a vector by detecting categoricals.
+ * A categorical will be represented as a vector consisting of occurrences of top K most common values of that feature
+ * plus occurrences of non top k values and a null indicator (if enabled).
+ * Non-categoricals will be removed.
+ *
+ * @param uid uid for instance
+ */
+class IDVectorizer
+(uid: String = UID[IDVectorizer])(implicit tti: TypeTag[ID])
+  extends SequenceEstimator[ID, OPVector](operationName = "IDVectorizer", uid = uid)
+    with PivotParams with CleanTextFun with SaveOthersParams
+    with TrackNullsParam with MinSupportParam with MaxCardinalityParams with OneHotFun {
+
+  private implicit val textStatsSeqEnc: Encoder[Array[TextStats]] = ExpressionEncoder[Array[TextStats]]()
+
+
+  def fitFn(dataset: Dataset[Seq[ID#Value]]): SequenceModel[ID, OPVector] = {
```

Comment:
@mweilsalesforce which parts of this fitFn were modified compared to `SmartTextVectorizer`?

Reply:
The main difference is in the args. `SmartTextVectorizerModelArgs` has a `hashingParams: HashingFunctionParams` ctor field. Here we don't need to do hashing etc.